pages/k8s/snaps.md

```diff
@@ -0,0 +1,189 @@
+  Note: Any files used by the service, such as certificate files, must be
+  placed within the /root/ directory to be visible to the service. This
+  limitation allows us to run a few of the services in a strict confinement
+  mode that offers better isolation and security.
+  </p>
+</div>
+
+
+After configuring, restart the service and you should see it running:
+
+```bash
+sudo service snap.kube-apiserver.daemon restart
+systemctl status snap.kube-apiserver.daemon
+```
+```no-highlight
+● snap.kube-apiserver.daemon.service - Service for snap application kube-apiserver.daemon
+   Loaded: loaded (/etc/systemd/system/snap.kube-apiserver.daemon.service; enabled; vendor preset: enabled)
+   Active: active (running) since Fri 2017-09-01 16:02:33 UTC; 6s ago
+   ...
+```
+
+### Configuration
+
+The keys and values for `snap set` map directly to arguments that you would
+normally pass to the service. You can view a list of arguments by invoking the
+service directly, e.g. `kube-apiserver -h`.
+
+For configuring the snaps, drop the leading dashes and pass them through
+`snap set`. For example, if you want kube-apiserver to be invoked like this:
+
+```bash
+kube-apiserver --etcd-servers https://172.31.9.254:2379 --allow-privileged
+```
+
+You would configure the snap like this:
+
+```bash
+snap set kube-apiserver etcd-servers=https://172.31.9.254:2379 allow-privileged=true
+```
+
+Note, also, that we had to specify a value of true for `allow-privileged`. This
+applies to all boolean flags.
+
+Config values are persisted individually, so if you later call
+`snap set kube-apiserver allow-privileged=false`, the value for `etcd-servers`
+will remain unchanged. You can remove an existing configuration by setting it
+to null:
+
+```bash
+snap set kube-apiserver allow-privileged=null
+```
+
+## Going further
+
+Want to know more? Here are a couple good things to know:
+
+If you're confused about what `snap set ...` is actually doing, you can read
+the snap configure hooks in `/snap/<snap-name>/current/meta/hooks/configure` to
+see how they work.
+
+The configure hook creates an args file at `/var/snap/<snap-name>/current/args`.
+This contains the actual arguments that get passed to the service by the snap:
+```
+--cert-dir "/root/certs"
+--etcd-cafile "/root/certs/ca.crt"
+--etcd-certfile "/root/certs/client.crt"
+--etcd-keyfile "/root/certs/client.key"
+--etcd-servers "https://172.31.9.254:2379"
+--service-cluster-ip-range "10.123.123.0/24"
+```
+
+<div class="p-notification--information">
+  <p class="p-notification__response">
+  Note: While you can technically bypass `snap set` and edit the args file
+  directly, it's best not to do so. The next time the configure hook runs, it
+  will obliterate your changes. This can occur not only from a call to
+  snap set` but also during a background refresh of the snap.
+  </p>
+</div>
+
+The source code for the snaps can be found here:
+<https://github.com/juju-solutions/release/tree/rye/snaps/snap>
```

Comment:
This URL is no longer correct. The snap code has moved to launchpad:

https://launchpad.net/snap-kubectl
https://launchpad.net/snap-kubeadm
https://launchpad.net/snap-kube-apiserver
https://launchpad.net/snap-kube-controller-manager
https://launchpad.net/snap-kube-scheduler
https://launchpad.net/snap-kubelet
https://launchpad.net/snap-kube-proxy